app/controllers/posts_controller.rb

```diff
@@ -10,5 +8,29 @@ def show
     @author = User.find(params[:user_id])
     @author_posts = @author.posts
     @post = @author_posts.find(params[:id])
+    @like = Like.new
+    @comment = Comment.new
+  end
+
+  def new
+    @author = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_data)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] Kindly refactor the code so that the post is generated and created for the `current_user`, not the `@user` from the route.

Reply:
Tky for the observation, I change it
Solved